File: cpp/leetcode/轮转数组/1.py
from typing import List

class Solution:
    def rotate(self, nums: List[int], k: int) -> List[int]:
        n = len(nums)
        if n==0:
            return []

        k %= n
        if k==0:
            return []

        def reverse(left: int, right: int) -> None:
            while left < right:
                nums[left], nums[right] = nums[right], nums[left]
                left += 1
                right -= 1

        reverse(0, n-1)
        reverse(0, k-1)
        reverse(k, n-1)

        return nums

if __name__ == "__main__":
    solution = Solution()
    nums = [1, 2, 3, 4, 5, 6, 7]
    result = solution.rotate(nums, 3)
    print(result)

File: cpp/leetcode/轮转数组/1.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    vector<int> rotate(vector<int>& nums, int k) {
        int n = nums.size();
        if (n == 0) return {};

        k %= n;
        if (k == 0) return {};

        reverse(nums.begin(), nums.end());
        reverse(nums.begin(), nums.begin() + k);
        reverse(nums.begin() + k, nums.end());

        return nums;
    }
};

int main() {
    Solution solution;
    vector<int> nums = {1, 2, 3, 4, 5, 6, 7};
    vector<int> result = solution.rotate(nums,3);
    cout << "[";
    for (const auto& res : result) {
        cout << res;
        if (&res != &result.back()) {
            cout << ", ";
        }
    }
    cout << "]" << endl;
}